scalding-core/src/main/scala/com/twitter/scalding/typed/KeyedList.scala

```diff
@@ -303,6 +303,19 @@ trait KeyedListLike[K, +T, +This[K, +T] <: KeyedListLike[K, T, This]]
 
   /** For each key, give the number of values */
   def size: This[K, Long] = mapValues { x => 1L }.sum
+
+  /**
+   * For each key, give the number of unique values. WARNING: May OOM.
+   * This assumes the values for each key can fit in memory.
+   */
+  def distinctSize: This[K, Long] = toSet[T].mapValues(_.size)
+
+  /**
+   * For each key, remove duplicate values. WARNING: May OOM.
+   * This assumes the values for each key can fit in memory.
+   */
+  def distinctValues[B >: T]: This[K, B] = toSet[B].flattenValues
```

Comment:
actually, can't we do the same trick here so that we always return `This[K, T]`? Since keyedlistlike is covariant in the value, this is just as good, and it simplifies the use case for the caller.